graphql_compiler/schema_transformation/split_query.py

```diff
@@ -0,0 +1,430 @@
+                                   (source field name, sink field name)
+            type_name_to_schema_id: Dict[str, str], mapping name of each type to the id of the
+                                    schema it came from
+            root_query_node: SubQueryNode, its query_ast and child_query_connections will be
+                             modified
+        """
+        self.type_info = type_info
+        self.edge_to_stitch_fields = edge_to_stitch_fields
+        self.type_name_to_schema_id = type_name_to_schema_id
+        self.root_query_node = root_query_node
+
+    def enter_Field(self, node, key, parent, path, *args):
+        """Check for split at the current field, creating a new SubQueryNode if needed.
+
+        If there is a new split at this field as clued by a @stitch directive, a new AST
+        will be created from the current branch, and the branch will be removed from the AST
+        being visited. If the property field used in the stitch is not present, it will be
+        added to the new AST. A new SubQueryNode will be created around this new AST, and
+        be added to the list of children of root_query_node. The rest of the branch will
+        not be visited.
+
+        Args:
+            node: Field that is being visited
+            key: int, index of the current node in the parent list
+            parent: List[Union[Field, InlineFragment]], containing all other fields or type
+                    coercions in this selection
+            path: List[Union[int, str]], listing the attribute names or list indices used to
+                  index into the AST, starting from the root, to reach the current node
+        """
+        self._check_directives_supported(node.directives)
+
+        # Get root vertex field name and selection set of the current branch of the AST
+        child_type_name, child_selection_set = \
+            self._get_child_root_vertex_field_name_and_selection_set(node)
+
+        if self._try_get_stitch_fields(node) is None:
+            # The type at the end of the edge doesn't cross schemas, check its schema id
+            self._check_or_set_schema_id(child_type_name)
+            return
+
+        if child_selection_set is None:
+            raise SchemaStructureError(
+                u'The edge "{}" is unexpectedly a property field with no further selection '
+                u'set, but edges marked with @stitch must be vertex fields, connecting '
+                u'vertices between schemas.'.format(node)
+            )
+
+        parent_field_name, child_field_name = self._try_get_stitch_fields(node)
+
+        # Check for existing field in parent
+        parent_field, parent_field_index = try_get_ast_and_index(
+            parent, parent_field_name, ast_types.Field
+        )
+        # Process parent field, and get parent field path
+        if parent_field is None:  # No existing source property field
+            # Change current field to stitch's source property field
```

Comment:
This is currently bugged -- if there is a normal vertex field before this cross schema vertex field, then this step will cause a property field to come after a vertex field in the selection, which is not allowed. Working on the nicest way to solve this issue.

Reply:
Solved by adding new property field between all existing property fields and existing vertex fields.